pkg/services/reputation/common/deps.go

```diff
@@ -29,7 +29,7 @@ type Writer interface {
 	// Close operation.
 	//
 	// Write must not be called after Close.
-	Write(reputation.Trust) error
+	Write(Context, reputation.Trust) error
```

Comment:
Why do we need  `Context`? We have `WriterProvider` for this purpose.

Reply:
since we try to use one router for `local` and `intermediate` purposes and - on the other hand - we want to use our local(mean "that repository") `Trust` struct, I decided to have some generalized "context" of the execution. if it is for intermediate usage - it must contain `Epoch()` and `I()` data, and intermediate controller must pass it

Reply:
Initially I asked why the `Context` in the signature when it is passed through `InitWriter` in expected way?

Reply:
got your question wrong.

i think that ctxs for `InitWriter` and `Write` are too different logically and in general *theoretically* may differ in practice(e.g. `passedRoute` may be same(received from one node through the same route) but received trusts are for different iterations)

there may be too much optional information if passed all through one ctx(may be just std `ctx`, may contain `passedRoute`, may have `Epoch`, may have `Iteration` and all this is optional). found it too complex

Reply:
and, by the way, think it is more traditional to pass context to func that communicates via network

Reply:
moved all ctx logic to `Init...` funcs. make `Write` methods ctx insensitive